XbSymbolDatabase.h

```diff
@@ -415,17 +419,26 @@ void XbSymbolSetOutputMessage(xb_output_message_t message_func);
 /// <param name="library_str">Name of the library in string.</param>
 /// <param name="library_flag">Name of the library in flag.</param>
 /// <param name="symbol_str">Name of the library in symbol string.</param>
-/// <param name="address">TBD.</param>
+/// <param name="address">Return xbox's virtual address.</param>
 /// <param name="revision">Found with specific revision.</param>
-typedef void (*xb_symbol_register_t)(const char* library_str, uint32_t library_flag, const char* symbol_str, uint32_t address, uint32_t revision);
+typedef void (*xb_symbol_register_t)(const char* library_str, uint32_t library_flag, const char* symbol_str, xbaddr address, uint32_t revision);
+
+/// <summary>
+/// To scan symbols in runtime xbe usage.
+/// </summary>
+/// <param name="xbeData">Starting point of xbe memory address.</param>
+/// <param name="register_func">Callback register function to be call for any detected symbols.</param>
+/// <returns>Only return false if something is not valid.</returns>
+bool XbSymbolScanRunTime(const void* xbeData, xb_symbol_register_t register_func);
+#define XbSymbolScan XbSymbolScanRunTime
 
 /// <summary>
-/// To scan symbols for Xbe's file.
+/// To scan symbols in Xbe's file image.
 /// </summary>
 /// <param name="xbeData">Starting point of xbe memory address.</param>
 /// <param name="register_func">Callback register function to be call for any detected symbols.</param>
 /// <returns>Only return false if something is not valid.</returns>
-bool XbSymbolScan(const void* xbeData, xb_symbol_register_t register_func);
+bool XbSymbolScanFileImage(const void* xbeData, xb_symbol_register_t register_func);
```

Comment:
What does this do? An Image can be addressed in various ways (Raw / Virtual).

When I read the PR description, I thought this would just load an XBE from file path. But this seems to do something else? So this looks poorly named.

Reply:
`Pointer to Raw image` is what this function do. Plus this function provide a complete scan than XbSymbolScanRunTime function which only check which section is loaded then scan those only.